src/config.ts

```diff
@@ -77,3 +87,89 @@ export const setConfig = (newConfig: ConfigOpts): Config => {
     ...newConfig,
   });
 };
+
+/**
+ * Get the provider and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetProvider = (opts?: ConfigOpts): ethers.providers.Provider => {
+  const c = getConfig(opts);
+  if (!c.provider) throw MissingProvider;
+  return c.provider;
+};
+
+/**
+ * Get the signer and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetSigner = (opts?: ConfigOpts): ethers.Signer => {
+  const c = getConfig(opts);
+  if (!c.signer) throw MissingSigner;
+  return c.signer;
+};
+
+/**
+ * Get the store and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetStore = (opts?: ConfigOpts): StoreInterface => {
+  const c = getConfig(opts);
+  if (!c.store) throw MissingStore;
+  return c.store;
+};
+
+/**
+ * Get the currentFromId and if undefined, throw.
+ * @param opts overrides for the current configuration.
+ */
+export const requireGetCurrentFromId = (opts?: ConfigOpts): string => {
+  const c = getConfig(opts);
+  if (!c.currentFromId) throw MissingUser;
+  return c.currentFromId;
+};
+
+/**
+ * Get the queue.  Since this is a required field, this is a plain getter.
+ * @param opts overrides for the current configuration.
+ */
+export const getQueue = (opts?: ConfigOpts): QueueInterface => {
+  const c = getConfig(opts);
+  return c.queue;
+};
+
+/**
+ * Get the contracts.  Since this is a required field, this is a plain getter.
+ * @param opts overrides for the current configuration.
+ */
+export const getContracts = (opts?: ConfigOpts): Contracts => {
+  const c = getConfig(opts);
+  return c.contracts;
+};
+
+/**
+ * Get the full configuration. Use if you need to guarantee that >1 configs are set
+ * @param requiredConfigs: list of config strings
+ * @param opts
+ */
+export const requireGetConfig = (requiredConfigs?: Array<string>, opts?: ConfigOpts): Config => {
+  const c = getConfig(opts);
```

Comment:
This comment is the only one that means something to me. When I was reading the code above which contain two lines, I overlooked the short abbreviations. But here when I am reading through the code that has more involved logic this short abbreviation kills the mental flow and readability for me.

I know that this is something small but can we please not use short abbreviations, at least, when there are more than two lines of code?

❤️ 🙏 por favor 🙏 ❤️

Reply:
sure thing 😸